### PR TITLE
add role_arn to support Service Role in CloudFormation

### PR DESCRIPTION
**Pull request by ssummer3**
##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME
module cloudformation

##### ANSIBLE VERSION
```
ansible 2.2.0.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/etc/ansible/library']
```

##### SUMMARY
Add `role_arn` to support [AWS CloudFormation Service Role](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/using-iam-servicerole.html)

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @ssummer3. To the current maintainers, @tedder, @ryansb please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
Thanks @ssummer3 for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by ssummer3** [comment]
Fails due to timeout of a test on windows? What revision needs to be made?

**Comment by tedder** [comment]
I have this tab open and will test it in the next few days.

File: cloud/amazon/cloudformation.py [hunk 1]
```diff
@@ -87,11 +87,18 @@
     choices: [ json, yaml ]
     required: false
     version_added: "2.0"
+  role_arn:
+    description:
+    - The role that AWS CloudFormation assumes to create the stack. [AWS CloudFormation Service Role](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/using-iam-servicerole.html)
```
Comment (ryansb) [inline cloud/amazon/cloudformation.py:92]: Markdown links don't work in docs, this has to be `U(http:// .......)` instead.

**Comment by ryansb** [comment]
Works for me :ship:ing it

**Pull request closed without merging**